spec/rubocop/cop/rails/squished_heredocs_spec.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Rails::SquishedHeredocs, :config do
+  subject(:cop) { described_class.new(config) }
+
+  it 'registers an offense' do
+    expect_offense(<<~RUBY)
+      <<~SQL
+      ^^^^^^ Use `<<~SQL.squish` instead of `<<~SQL`.
+        SELECT * FROM table;
+      SQL
+    RUBY
```

Comment:
All of the tests here inspect a heredoc on its own, but it would be good to also cover a heredoc defined as an inline method argument, and in particular when it's followed by another argument:

```ruby
execute(<<~SQL, "Post Load")
  SELECT * FROM posts
    WHERE post_id = 1
SQL

Reply:
This is a very very good point, which I did not think about before.

I will look into it, it will make the development for the cop a bit more complex for me 😉

Reply:
I did spend some time now, trying to find a solution for the `execute(<<~SQL, "Post Load")` code, but didn't find on yet.

I saw, that the base cop [`Layout/HeredocArgumentClosingParenthesis`[(https://github.com/rubocop-hq/rubocop/blob/master/lib/rubocop/cop/layout/heredoc_argument_closing_parenthesis.rb) has some logic around handling these types of heredocs, but looking at the code - it is quite complex and I guess grew over time as well.

I did add a check, to not check Heredocs that are used directly as a method argument, although the check might not be 100% correct I bet :|

@eugeneius What do you say to the newest implementation?

Reply:
I think `Layout/HeredocArgumentClosingParenthesis` is tricky because it needs to move code around the heredoc body. This cop doesn't need to modify the body, so it shouldn't be as difficult to handle all heredocs. You are right that it's a bit more complex, though 🙂 

The new approach seems fine to me, although I think there should be a test to demonstrate that it doesn't mangle the code I posted above. You could use `expect_no_offenses` to do that.